ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageSievingInputStream.java

```diff
@@ -102,7 +106,15 @@ public MessageSievingInputStream(InputStream inStream, List<MessageInfo> message
       // @todo: (say, due to corruption). This can help avoid a copy.
       Message msg = new Message(msgInfo, new ByteArrayInputStream(Utils.readBytesFromStream(inStream, msgSize)));
       logger.trace("Read stream for message info " + msgInfo + "  into memory");
-      sieve(msg, msgStreamList, bytesRead);
+      if (msg.getMessageInfo().isDeleted()) {
```

Comment:
minor: if we check these conditions before, maybe we can just read/skip the stream for the given length (instead of creating a byte array and then discarding it)

Reply:
Addressed this.